pkg/storage/chunk/fetcher/fetcher.go

```diff
@@ -195,6 +201,32 @@ func (c *Fetcher) FetchChunks(ctx context.Context, chunks []chunk.Chunk, keys []
 		level.Warn(log).Log("msg", "error process response from cache", "err", err)
 	}
 
+	var fromCacheL2 []chunk.Chunk
+
+	if c.l2CacheHandoff > 0 {
+		// Fetch missing from L2 chunks cache
+		missingL1 := make([]string, 0, len(missing))
+		for _, m := range missing {
+			chunkKey := c.schema.ExternalKey(m.ChunkRef)
+			missingL1 = append(missingL1, chunkKey)
+		}
+
+		cacheHitsL2, cacheBufsL2, _, err := c.cachel2.Fetch(ctx, missingL1)
```

Comment:
can skip checking l2 when the chunk's `from` is earlier than the the `l2` handoff, but not the reverse b/c chunks in `l1` aren't immediately evicted after this period elapses